artifactory/commands/transferconfig/transferconfigmerge.go

```diff
@@ -0,0 +1,373 @@
+		s := ""
+		result = true
+		if sourceProject.DisplayName != targetProject.DisplayName {
+			s = tcc.addToDifferentProperty(s, "Display Name")
+		}
+		if sourceProject.ProjectKey != targetProject.ProjectKey {
+			s = tcc.addToDifferentProperty(s, "Project Key")
+		}
+		if result {
+			if sourceProject.Description != targetProject.Description {
+				s = tcc.addToDifferentProperty(s, "Description")
+			}
+			if sourceProject.StorageQuotaBytes != targetProject.StorageQuotaBytes {
+				s = tcc.addToDifferentProperty(s, "Storage Quota Bytes")
+			}
+			if sourceProject.SoftLimit != nil && targetProject.SoftLimit != nil {
+				if *sourceProject.SoftLimit != *targetProject.SoftLimit {
+					s = tcc.addToDifferentProperty(s, "Soft Limit")
+				}
+			}
+			if sourceProject.SoftLimit == nil || targetProject.SoftLimit == nil {
+				if sourceProject.SoftLimit != nil || targetProject.SoftLimit != nil {
+					s = tcc.addToDifferentProperty(s, "Soft Limit")
+				}
+			}
+			if !tcc.checkIfSameAdminPrivilige(sourceProject.AdminPrivileges, targetProject.AdminPrivileges) {
+				s = tcc.addToDifferentProperty(s, "Admin Privileges")
+			}
+		}
+		if s != "" {
+			conflict := tcc.newConflict(sourceProject.DisplayName, targetProject.DisplayName, sourceProject.ProjectKey, targetProject.ProjectKey, s)
+			conflicts = append(conflicts, conflict)
+			return conflicts, true
+		}
+
+	}
+	return conflicts, result
+}
+
+func (tcc *TransferConfigCommand) checkIfSameAdminPrivilige(source, target *accessServices.AdminPrivileges) bool {
+	if source == nil && target == nil {
+		return true
+	}
+
+	if source == nil || target == nil {
+		return false
+	}
+
+	// if source and target Admin priviliges are not nil then we have to check all pointer admin privilige have
+	manageMember := tcc.checkIfsameBoolPointer(source.ManageMembers, target.ManageMembers)
+	manageResouce := tcc.checkIfsameBoolPointer(source.ManageResources, target.ManageResources)
+	indexResouce := tcc.checkIfsameBoolPointer(source.IndexResources, target.IndexResources)
+	return (manageMember && manageResouce && indexResouce)
+
+}
+
+func (tcc *TransferConfigCommand) checkIfsameBoolPointer(source, target *bool) bool {
+	if source != nil && target != nil {
+		if *source != *target {
+			return false
+		}
+	}
+	if source == nil || target == nil {
+		if source != nil || target != nil {
+			return false
+		}
+	}
+	return true
+}
+
+func (tcc *TransferConfigCommand) addToDifferentProperty(s, property string) string {
+	if s == "" {
+		s = property
+		return s
+	}
+	s = s + ", " + property
+	return s
+}
+
+func (tcc *TransferConfigCommand) tryPing(targetServicesManager artifactory.ArtifactoryServicesManager) error {
+	_, err := targetServicesManager.Ping()
+	return err
+}
+
+func (tcc *TransferConfigCommand) validateMergeArtifactoryServers(targetServicesManager artifactory.ArtifactoryServicesManager, sourceArtifactoryVersion string, minRequiredVersion string) (bool, error) {
```

Comment:
Let's have all comments in this function start with an upper case letter.
Let's also add a description for this function.